<!--
  ~ Copyright 2024 Robotec.ai
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~     http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  -->

<launch>
    <arg name="dino_weights_path" default = ""/>
    <arg name="sam_weights_path" default = ""/>
    <arg name="image_path" />

    <node pkg="rai_perception" exec="grounding_dino" name="grounding_dino" output="screen">
        <param name="weights_path" value="$(var dino_weights_path)" />
    </node>

    <node pkg="rai_perception" exec="grounded_sam" name="grounded_sam" output="screen">
        <param name="weights_path" value="$(var sam_weights_path)" />
    </node>

    <node pkg="rai_perception" exec="talker" name="example_client" output="screen">
      <param name="image_path" value="$(var image_path)" />
    </node>
</launch>
